cellxgene_schema_cli/tests/test_schema_compliance.py

```diff
@@ -632,6 +633,22 @@ def test_assay_ontology_term_id(self, validator_with_adata, assay_ontology_term_
         ]
         assert validator.errors == [self.get_format_error_message(error_message_suffix, error)]
 
+    def test_assay_ontology_term_id__as_categorical(self, validator_with_visium_assay):
+        """
+        Formally, assay_ontology_term_id is expected to be a categorical variable of type string. However, it should work for categorical dtypes as well.
+        """
+        validator: Validator = validator_with_visium_assay
+
+        # check encoding as string
+        validator._validate_obsm()
```

Comment:
this does not exercise your changes. `validator._check_spatial` would be better. Alternatively you can call `validator._validate_spatial_cell_type_ontology_term_id`, and `validator._validate_spatial_tissue_position` to reduce the test area.

Reply:
thanks @Bento007 good catch. This was a tab-complete oversight on my part. Updated to `check_spatial`.